src/common/sendpacket.c

```diff
@@ -458,8 +495,15 @@ sendpacket(sendpacket_t *sp, const u_char *data, size_t len, struct pcap_pkthdr
                 retcode, len);
         sp->trunc_packets ++;
     } else {
+        #ifndef HAVE_LIBXDP
```

Comment:
This solution may seems to be strange, however we need to handle the following cases:
1. dependencies of HAVE_LIBXDP are not installed, in this case everything should be the same as it was before
2. In case HAVE_LIBXDP is defined we have two additional cases:
2.1. Packet processing doesn't use LIBXDP (it is possible because of the hierarchical sendpacket selection), in this case we still want to use the old logic.
2.2. Packet processing uses LIBXDP, due to batch processing  we should count these at a different place

Reply:
There is another solution where I eliminate the include guard, but in that case if(sp->handle_type != ...) should be calculated. I'm not sure whether I can introduce this in cases where LIBXDP is not intended to be used.

Reply:
You are sure about case 2.1.? As far as I understand it about the include/define magic, we cant have more packet sending option than one. So forcing PF_PACKET for example sont unset HAVE_LIBXDP?

Reply:
In case you force PF_PACKET it will unset HAVE_LIBXDP. However if you have all the dependencies installed for all packet processing methods and you DO NOT force any method, then the hierarchical include/define stuff will select the first element without unsetting the others (e.g. HAVE_LIBXDP). In this case both PF_PACKET and HAVE_LIBXDP is defined. 

I think due to the #if ... #elseif   .... #elseif structure of sendpacket_t selection in sendpacket_open function, there is no other solution, it would be better to be honest with you to forget the include/define structure and let the user select the packet processing method via the tcpreplay interface like   tcpreplay -i veth0 --method=xxx